Tests for plotting functions and methods, other increases in test coverage

ulupo opened the pull request:
**Reference issues/PRs**
Addresses several remaining gaps in coverage (but still not the ones indicated in #182).

**Types of changes**
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Description**
- Adds tests for all `PlotterMixins` or other classes containing `plot` methods. These are just **dummy tests** implementing trivial calls to `fit_transform_plot` or `plot`.
- Updates `.coveragerc` file so that `gtda.images` is included in coverage reports, `gtda.utils._docs` and `gtda.utils.intervals` are excluded, and all test directories are excluded.
- Adds simple tests for `Filtering`.
- Adds simple shape tests for `BettiCurve` and `PersistenceLandscape`.
- Minor typo fixes and other code style improvements.

**Checklist**
- [x] I have read the [guidelines for contributing](https://giotto-ai.github.io/gtda-docs/latest/contributing/#guidelines).
- [x] My code follows the code style of this project. I used `flake8` to check my Python changes.
- [ ] My change requires a change to the documentation.
- [ ] ~~I have updated the documentation accordingly.~~
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed. I used `pytest` to check this on Python tests.

Commit:
Bump version to 0.2.0, add release notes

Commit:
Add Adelie as contributor

Commit:
Add Mapper color bug

Commit:
Update ccache

Commit:
Add @ammermar as a contributor in release notes

Commit:
Add @ammedmar as a contributor in release notes

Commit:
List all classes in images subpackage

Commit:
List all classes in images subpackage

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Add new lines before bullet lists in rst

Commit:
Add forgotten line break

Commit:
Attempt at improving LaTeX rendering in mapper_quickstart

Commit:
Fix broken URL in release notes

Commit:
Merge branch 'master' of https://github.com/giotto-ai/giotto-tda

Commit:
Add @rth to code authors and list of contributors for v0.2.0

Commit:
Add twine check to azure-pipelines.yml, attempt at fix to README

Commit:
Merge branch 'master' of https://github.com/giotto-ai/giotto-tda

Commit:
Fix duplicate link, not accepted by twine

Commit:
Fix typo and remove another duplicate

Commit:
Merge branch 'master' of https://github.com/giotto-ai/giotto-tda

Commit:
Remove images subpackage from code omitted from coverage

Commit:
Add tests for Filtering class

Commit:
Add tests in diagrams subpackage, linting

- Rename test_features -> test_features_representations
- Create parametrized dummy tests for fit_transform_plot method of transformers in representations.py
- Improve structure and range of tests in test_preprocessing in a similar way
- Create dummy tests of shape for BettiCurve and PersistenceLandscape
- Linting and code style changes elsewhere

Commit:
Add fit_transform_plot test for GraphGeodesicDistance

Commit:
Add fit_transform_plot tests in gtda.homology

Commit:
Fix typo in voids_on_the_plane notebook

Commit:
Add fit_transform_plot tests for images subpackage

Commit:
Add fit_transform_plot tests for point_clouds subpackage

Commit:
Add test for SlidingWindow.plot

Commit:
Linting

Commit:
Fix flake8 issues

wreise commented on gtda/diagrams/representations.py:
```diff
@@ -179,51 +179,51 @@ def plot(self, Xt, sample=0, homology_dimensions=None):
-            },
-            "plot_bgcolor": "white"
-        }
+        layout = dict(
+            xaxis1=dict(
+                title="Filtration parameter",
+                side="bottom",
+                type="linear",
+                ticks="outside",
+                anchor="x1",
+                showline=True,
+                zeroline=True,
+                showexponent="all",
+                exponentformat="e"
+            ),
+            yaxis1=dict(
+                title="Betti number",
+                side="left",
+                type="linear",
+                ticks="outside",
+                anchor="y1",
+                showline=True,
+                zeroline=True,
+                showexponent="all",
+                exponentformat="e"
+            ),
+            plot_bgcolor="white"
+        )
```
I don't know much about this. Could I ask why would we prefer `dict()` over `{}`?

ulupo replied:
In several IDEs (like Pycharm), syntax colouring will be more helpful when using `dict()`, as it clearly separates the keywords from the values. Also, as far as I remember it is the way `plotly` documentation tends to prefer things done (for whatever reason).

wreise commented on gtda/diagrams/tests/test_features_representations.py:
```diff
@@ -9,38 +9,77 @@
 
-def test_pe_not_fitted():
-    pe = PersistenceEntropy()
+def test_not_fitted():
+    with pytest.raises(NotFittedError):
+        PersistenceEntropy().transform(X)
+
+    with pytest.raises(NotFittedError):
+        BettiCurve().transform(X)
+
+    with pytest.raises(NotFittedError):
+        PersistenceLandscape().transform(X)
+
+    with pytest.raises(NotFittedError):
+        HeatKernel().transform(X)
+
+    with pytest.raises(NotFittedError):
+        PersistenceImage().transform(X)
```
Yeah!

Commit:
Exclude gtda.utils._doc and gtda.utils.intervals from coveragerc

wreise commented on gtda/diagrams/tests/test_features_representations.py:
```diff
@@ -9,38 +9,77 @@
+
+
+@pytest.mark.parametrize('hom_dim_ix', [0, 1])
+def test_fit_transform_plot_1(hom_dim_ix):
```
Would you mind giving a more descriptive name, or a comment for future selves?
If i understand correctly, we plot a single homology dimension.

ulupo replied:
Some classes have the `homology_dimension_ix` parameter, others `homology_dimensions`. I could not find a short convincing name, but `test_fit_transform_plot_one_hom_dim` for this one, and `test_fit_transform_plot_single_many_hom_dims` for the next, could work with me.

wreise replied:
👍

wreise commented on gtda/diagrams/tests/test_features_representations.py:
```diff
@@ -9,38 +9,77 @@
+
+
+@pytest.mark.parametrize('hom_dims', [None, (0,), (1,), (0, 1)])
+def test_fit_transform_plot_2(hom_dims):
```
Same

ulupo commented on gtda/diagrams/representations.py:
```diff
@@ -179,51 +179,51 @@ def plot(self, Xt, sample=0, homology_dimensions=None):
-                        f'All homology dimensions must be in '
-                        f'self.homology_dimensions_ which is '
-                        f'{self.homology_dimensions_}. {dim} is not.')
+                        f"All homology dimensions must be in "
```
I'm starting to use these quotation marks consistently in sentence-like strings as they are safe when apostrophes have to be used inside as well.

Commit:
Implement @wreise's suggestions

- Rename some test functions
- Implement a solution to the automatic tab opening problem, based on setting the default value of plotly.io.renderers in test files

Commit:
Upload coverage report only from macOS job

Commit:
Fix typo

Commit:
Exclude test folders from coverage reports
